File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/ToastHandler.java
package closer.vlllage.com.closer.handler.helpers;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;
import android.widget.Toast;

import closer.vlllage.com.closer.pool.PoolMember;

public class ToastHandler extends PoolMember {

    public void show(@StringRes int message) {
        Toast.makeText($(ApplicationHandler.class).getApp(), message, Toast.LENGTH_SHORT).show();
    }

    public void show(@NonNull String message) {
        Toast.makeText($(ApplicationHandler.class).getApp(), message, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/ToastHandler.kt
package closer.vlllage.com.closer.handler.helpers

import androidx.annotation.StringRes
import android.widget.Toast

import com.queatz.on.On

class ToastHandler constructor(private val on: On) {

    fun show(@StringRes message: Int, long: Boolean = false) {
        Toast.makeText(on<ApplicationHandler>().app, message, duration(long)).show()
    }

    fun show(message: String, long: Boolean = false) {
        Toast.makeText(on<ApplicationHandler>().app, message, duration(long)).show()
    }

    private fun duration(long: Boolean) = when (long) {
        true -> Toast.LENGTH_LONG
        false -> Toast.LENGTH_SHORT
    }
}
